scripts/app.js

```diff
@@ -131,3 +134,39 @@ function renderWallpaper(player) {
   ctx.drawImage(images.background, 0, 0);
   ctx.drawImage(player, 0, 0);
 }
+
+exportButton.addEventListener("click", () => {
+  const template = {
+    background: convertImageToBase64(images.background),
+    player: convertImageToBase64(images.player),
+    hat: convertImageToBase64(images.hat)
+  };
+
+  const jsonData = JSON.stringify(template);
+  download(jsonData, "wp_template.json");
+});
+
+importButton.addEventListener("change", (event) => {
+  const file = event.target.files[0];
+  const reader = new FileReader();
+  reader.onload = (event) => {
+    const jsonData = event.target.result;
+    const template = JSON.parse(jsonData);
+
+    images.background = new Image();
+    images.player = new Image();
+    images.hat = new Image();
+
+    images.background.onload = () => {
```

Comment:
Quite a large ladder of callbacks. Use async await or just make sure it doesn't grow any more in future